### ISSUE TITLE
Youtube App Video Auto Pause 

### ISSUE THREAD

**Issue by perusudroid** [eneim/toro#439]
What kind of issue is this?

- In YouTube player app, if i place view above player the player i get Unauthorised overlay error message. After removing it, i have just the player alone in the layout removing all other views, am not getting the error message in toast but the player get paused. 

Once the player is automatically paused, even after the activity or fragment destroyed and recreated also, can't able to play the video as it automatically pauses.

If the video is auto paused, it not only affects the activity or fragment where it is implemented, but also the other activity or fragment where the player is implemented.

How i could completely release the player?? I tried to release the player in onDestroy() of the fragment. But still i couldn't figure out what is actually wrong!. My app is Live with toro youtube player. Please help to resolve it.

**Comment by eneim** [comment]
Can you explain why you want to completely release the player? In my demo if the RecyclerView that use Youtube player is detached, everything will be released.

Still not get all your issue. Please explain it in steps: what do you want to archive, what is the trouble now? what is the “thing” that is the main issue in your app?

**Comment by perusudroid** [comment]
Hi, here is the link to my app,
https://play.google.com/store/apps/details?id=com.perusudroid.myhero

I have used your Demo for Playing YouTube videos in RecyclerView, Thanks for sharing such a wonderful demo app. Without which i might not able to used it in my app. Please see in the app, after scrolling down, somewhere the app stops playing the video and then actually it never release at all. I mean, untill the app is reopened that issue remains, You could see it live. Thanks..

**Comment by eneim** [comment]
@perusudroid I cannot reproduce the issue. Can you take a record of your screen and send me?

**Comment by perusudroid** [comment]
Please look at the video i have attached.
If it is an Un-Authorized overlay toast will popup. But when i scroll and switch to next video, video autopauses.
https://drive.google.com/file/d/13hWpV3FdaUKIOQZmr1qigXj460Ab3cxt/view?usp=sharing

**Comment by venkateshimmidi** [comment]
@perusudroid is the issue solved?? Got any solution?? I am also facing the same issue, plz help me out if any solution.

**Comment by perusudroid** [comment]
Hi there,

Nope. I am in search of other good library to replace it.

**Comment by venkateshimmidi** [comment]
thanks for the quick response. Pls let me know if any thing found. I am also searching for other library.

**Comment by perusudroid** [comment]
For which purpose you use this library?. Is your app @ playstore?

**Comment by venkateshimmidi** [comment]
No not yet in playstore. Still in development. We have a requirement of playing youtube videos like youtube app.

**Comment by eneim** [comment]
@perusudroid How often the issue happens to your app? Can you share the xml of the Card you put YouTube fragment to? I'm restarting the work with YouTube this week, maybe I can see a work around. As said, YouTube Player SDK is not friendly enough to have a fully working implementation.

**Comment by venkateshimmidi** [comment]
can i use any other player instead of Exoplayer and youtube player with toro??

**Comment by eneim** [comment]
@venkateshimmidi What do you mean by other player?

If you want to play YouTube videos, you need either YouTube Android Player API or its IFrame API (there is a wrapper for Android).

If you want to play normal Video using player other than ExoPlayer, it is Ok. You will need to write your own version of Helper for it. Check my legacy demo to see how I build it for Android's MediaPlayer API.

**Comment by venkateshimmidi** [comment]
> @perusudroid How often the issue happens to your app? Can you share the xml of the Card you put YouTube fragment to? I'm restarting the work with YouTube this week, maybe I can see a work around. As said, YouTube Player SDK is not friendly enough to have a fully working implementation.

On scrolling the page top to bottom and from bottom to top, continuously, then on playing the youtube video then video video gets started after 1 sec it stops and  gets the error msg in logcat like.."**YouTubeAndroidPlayerAPI: YouTube video playback stopped due to the player's view being too small. The YouTubePlayerView is 0dp wide (minimum is 200dp) and 0dp high (minimum is 110dp).**"

**Comment by eneim** [comment]
@venkateshimmidi Please try to set a fix height (big enough) for your ``android:id="@+id/player_view"`` and also remove the ImageView on top of it and retry.

**Comment by venkateshimmidi** [comment]
yes i have tried with the given suggestion but still it is showing the same
error.

**Comment by venkateshimmidi** [comment]
I think there is a problem with the progress bar(but not sure). But there
is no option to hide it, it is internally coming from the library. Can you
help me with this?
